src/cpp/fastdds/publisher/DataWriterImpl.cpp

```diff
@@ -1462,11 +1462,7 @@ std::shared_ptr<IPayloadPool> DataWriterImpl::get_payload_pool()
         else
         {
             payload_pool_ = TopicPayloadPoolRegistry::get(topic_->get_name(), config);
-            if (!std::static_pointer_cast<ITopicPayloadPool>(payload_pool_)->reserve_history(config, false))
-            {
-                auto topic_pool = std::static_pointer_cast<ITopicPayloadPool>(payload_pool_);
-                TopicPayloadPoolRegistry::release(topic_pool);
-            }
+            std::static_pointer_cast<ITopicPayloadPool>(payload_pool_)->reserve_history(config, false);
```

Comment:
Should we at least do a `payload_pool_.reset()` when `reserve_history()` fails?

Reply:
Certainly!